src/components/SQFormDialog/SQFormDialogInner.tsx

```diff
@@ -183,8 +184,8 @@ function SQFormDialogInner({
           <span style={{paddingRight: '20px'}}>
             <SQFormButton
               title={tertiaryButtonText}
-              isDisabled={isTertiaryDisabled}
-              onClick={onTertiaryClick}
+              isDisabled={isTertiaryDisabled || !formikContext.isValid}
+              onClick={() => onTertiaryClick?.(formikContext)}
```

Comment:
Maybe this is how we want it...
if as a consumer don't want the form validitiy to be considered and just manage it through isTertiaryDisabled... then false may do things I dont want.
Would it be better to maybe do `isTertiaryDisabled ?? !formikContext.isVaild` ?

just a thought, I think the common use case is a submit the form and keep a modal open for another so maybe we don't care?

Reply:
Doing `isTertiaryDisabled ?? !formikContext.isVaild` makes it impossible to use the `isTertiaryDisabled` and the formikContext.isValid together. Most common use case is disabling the button when the modal is loading but we also want the button disabled with the form validation.

Reply:
Since the conditional is not a "one or the other" we need to break this out into a helper function. How about this?

```typescript
function getIsDisabled() {
  if (isTertiaryDisabled === true || !formikContext.isValid) {
    return true
    }
    
   return false;
}
```

Reply:
```
if(noValidation) {
  return Boolean(isTertiaryDisabled)
}

if (isTertiaryDisabled || !formContext.isValid) {
  return true
}

return false;

```

Reply:
Hrmmm....i'm leaning towards introducing a breaking change. This dialog's props API is already big, the tertiary button is adding 4 or 5 props to it.

We can eliminate the `noValidation`,  `showTertiaryButton` & `isTertiaryDisabled` props with a `tertiaryStatus` prop.

The prop can be a TS Union to capture all the possible states of the Tertiary button. While this is more verbose, I think it guards against impossible states (if we were to pass isTertiaryDisabled=true and noValidation={true}) and gives the consumer full control over the validation and rendering with one prop.

```typescript
// new Prop
type TertiaryStatus = 'HIDE_BUTTON' | 'NO_VALIDATION' | 'IS_DISABLED' | 'IS_ENABLED' | 'FORM_VALIDATION_ONLY' | 'IS_DISABLED_AND_FORM_VALIDATION'
```

Then we can use the status to derive our values

```typescript
// pseudo codez
showTertiary={status !== 'HIDE_BUTTON}
isDisabled={getIsDisabled()}

function getIsDisabled() {
   switch (status) {
      case 'IS_DISABLED':
         return true
      case 'FORM_VALIDATION_ONLY':
         return !formContext.isValid
      case 'IS_DISABLED_AND_FORM_VALIDATION':
         return status === 'IS_DISABLED' || !formContext.isValid)
       default
         return false
   }
}
```

Reply:
Seems reasonable with a default value of HIDE_BUTTON. We could also perhaps export an enum or something with those values for ease of use.

Reply:
Agreed. 
Set the default value to hide, and export the Union type to the consumers

Reply:
@SeanGroff I think I have adjusted the code to function as you and Branden have outlined above (and ill push it up soon for review) but the one thing I'm kind of confused about is the case in which the tertiaryStatus is 'IS_DISABLED_AND_FORM_VALIDATION' .

I interpreted this as a use case in which the user wants the button to use validation to determine if the button is disabled but also another boolean value such as when a query is loading, but wouldn't we need another prop for the user to put the additional boolean value? sorry for the questions!

Reply:
The naming from my pseudo code above probably isn't great so feel free to use better naming.

`IS_ENABLED` would be the equivalent to `isDisabled === false`
`IS_DISABLED` would be the equivalent to `isDisabled === true`
'NO_VALIDATION` would be the equivalent to `noValidation === true // do not run any validation of any kind`